network/netcore/src/transport/mod.rs

```diff
@@ -86,7 +87,7 @@ pub trait Transport {
         Self: Sized;
 
     /// Dials the given [`NetworkAddress`], returning a future for a pending outbound connection.
-    fn dial(&self, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error>
+    fn dial(&self, peer_id: PeerId, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error>
```

Comment:
i'm still not sure how i feel about this, since it doesn't really make sense for base transports. or for example the old peer-id exchange protocol. it really only makes sense for the libranet transport... this is asserting that we must always know the precise peer id for a remote peer for any transport, while in some hypothetical transports this could be exchanged in-band.

I suppose it's fine for now. we'll definitely have to revisit these decisions when/if we look at refactoring the Transport trait to support combinators again.

Reply:
I see what you're saying, but I don't think we should build something too general. If we only have situation when we're dialing a specific peer, then we should use that information :o 

worst case we can make it an option in the future?

Reply:
right, like i said, in the interest of time i'm fine with adding the `PeerId` arg. i think `Transport` trait could definitely use some love, but we'll have to investigate that later : )